--- conf/common.rmk	2009-06-18 03:16:51.154703480 +0600
+++ conf/common.rmk	2009-06-19 18:10:26.815260342 +0600
@@ -157,10 +157,7 @@
 %: util/grub.d/%.in config.status
 	./config.status --file=$@:$<
 	chmod +x $@
-grub-mkconfig_SCRIPTS = 00_header 30_os-prober 40_custom
-ifneq (, $(host_kernel))
-grub-mkconfig_SCRIPTS += 10_$(host_kernel)
-endif
+grub-mkconfig_SCRIPTS = 00_header 10_linux 10_freebsd 10_windows 30_os-prober 40_custom
 
 CLEANFILES += $(grub-mkconfig_SCRIPTS)
 
