--- amarok/src/playlistbrowser.cpp	2007-09-01 07:26:32.836087653 +0600
+++ amarok/src/playlistbrowser.cpp	2007-09-01 07:26:17.877239103 +0600
@@ -792,8 +792,9 @@ void PlaylistBrowser::loadDefaultSmartPl
     /********** All Collection **************/
     qb.initSQLDrag();
     qb.sortBy( QueryBuilder::tabArtist, QueryBuilder::valName );
+    qb.sortBy( QueryBuilder::tabYear, QueryBuilder::valName );
     qb.sortBy( QueryBuilder::tabAlbum, QueryBuilder::valName );
-    qb.sortBy( QueryBuilder::tabSong, QueryBuilder::valTrack );
+    qb.sortBy( QueryBuilder::tabSong, QueryBuilder::valTrack );
 
     item = new SmartPlaylist( m_smartDefaults, 0, i18n( "All Collection" ), qb.query() );
     item->setPixmap( 0, SmallIcon( Amarok::icon( "collection" ) ) );
