diff --git filelight/admin/configure.in.min filelight/admin/configure.in.min
index 0dfe378..5aa14b1 100644
--- filelight/admin/configure.in.min
+++ filelight/admin/configure.in.min
@@ -43,6 +43,11 @@ KDE_SET_PREFIX
 dnl generate the config header
 AM_CONFIG_HEADER(config.h) dnl at the distribution this done
 
+AC_PROG_CC
+AC_PROG_CXX
+AC_EXEEXT
+AC_OBJEXT
+
 dnl Checks for programs.
 AC_CHECK_COMPILERS
 AC_ENABLE_SHARED(yes)
