--- src/coverdisplay.cpp.old	2008-04-24 00:56:57.633888737 +0600
+++ src/coverdisplay.cpp	2008-04-24 00:59:07.357141307 +0600
@@ -381,7 +381,7 @@ void CoverDisplay::mousePressEvent(QMous
 void CoverDisplay::wheelEvent(QWheelEvent *event)
 {
 	if (areControlsShown()) {
-		if (event->orientation() == Qt::Vertical) {
+		if (event->orientation() == Qt::Horizontal) {
 			PlayerInformation *infos = PlayerInformation::instance();
 			if (infos->canSeek()) {
 				int deltaSeconds = 10 * (event->delta() > 0 ? 1 : -1);
