--- sixad/Makefile_old	2011-08-09 22:53:32.565244588 +0600
+++ sixad/Makefile	2011-08-09 22:56:07.812879836 +0600
@@ -31,12 +31,12 @@
 	@echo "Please run '[sudo] make install-system' now if you're installing from source"
 
 install-system:
-	@addgroup --system sixad || true
-	@chown :sixad $(DESTDIR)/var/lib/sixad/
+#	@addgroup --system sixad || true
+	@chown :users $(DESTDIR)/var/lib/sixad/
 	@chmod 775 -R $(DESTDIR)/var/lib/sixad/
 	@echo "Installation is Complete!"
-	@echo "You may want to add yourself to the sixad group now:"
-	@echo "$ [sudo] adduser YOUR-USER sixad"
+	@echo "You may want to add yourself to the users group now:"
+	@echo "$ [sudo] adduser YOUR-USER users"
 
 uninstall:
 	rm -f $(DESTDIR)/etc/default/sixad
