--- src/alicecore/alice_xhelper.orig	2007-12-01 14:09:40.000000000 +0100
+++ src/alicecore/alice_xhelper	2007-12-01 14:12:41.000000000 +0100
@@ -62,10 +62,10 @@
 /usr/bin/xsetroot -solid "#5971A5" -cursor_name left_ptr
 
 # Start the DCOP server
-/usr/bin/dcopserver
+`kde-config --prefix`/bin/dcopserver
 
 # Start the KWin
-/usr/bin/kwin --lock &
+`kde-config --prefix`/bin/kwin --lock &
 
 # Export the KDE_FULL_SESSION
 #KDE_FULL_SESSION=true
