diff -Naur diskman-0.9.7-org/src/alicecore/loperationdetail.h diskman-0.9.7/src/alicecore/loperationdetail.h
--- diskman-0.9.7-org/src/alicecore/loperationdetail.h	2007-11-19 14:49:26.000000000 +0100
+++ diskman-0.9.7/src/alicecore/loperationdetail.h	2008-04-27 11:32:11.000000000 +0200
@@ -22,6 +22,7 @@
 #ifndef LOPERATIONDETAIL
 #define LOPERATIONDETAIL
 
+#include <ctime>
 #include <qstring.h>
 #include <qvaluevector.h>
 
diff -Naur diskman-0.9.7-org/src/aliceui/lxapplication.cpp diskman-0.9.7/src/aliceui/lxapplication.cpp
--- diskman-0.9.7-org/src/aliceui/lxapplication.cpp	2007-11-19 14:49:35.000000000 +0100
+++ diskman-0.9.7/src/aliceui/lxapplication.cpp	2008-04-27 11:41:14.000000000 +0200
@@ -26,6 +26,7 @@
 #include <qapplication.h>
 
 #include <iostream>
+#include <cstdlib>
 
 #include "config.h"
 
diff -Naur diskman-0.9.7-org/src/diskman/qparted/device.cpp diskman-0.9.7/src/diskman/qparted/device.cpp
--- diskman-0.9.7-org/src/diskman/qparted/device.cpp	2007-11-19 14:49:38.000000000 +0100
+++ diskman-0.9.7/src/diskman/qparted/device.cpp	2008-04-27 11:49:54.000000000 +0200
@@ -17,6 +17,7 @@
 
 #include "qparted/sys/sort.h"
 #include "qparted/utils.h"
+#include <algorithm>
 
 #include "qparted/device.h"
 
diff -Naur diskman-0.9.7-org/src/diskman/qparted/partition.cpp diskman-0.9.7/src/diskman/qparted/partition.cpp
--- diskman-0.9.7-org/src/diskman/qparted/partition.cpp	2007-11-19 14:49:38.000000000 +0100
+++ diskman-0.9.7/src/diskman/qparted/partition.cpp	2008-04-27 11:55:19.000000000 +0200
@@ -17,6 +17,7 @@
 
 #include "qparted/utils.h"
 #include "qparted/sys/sort.h"
+#include <algorithm>
 
 #include "qparted/partition.h"
 
diff -Naur diskman-0.9.7-org/src/diskman/qparted/thread2.cpp diskman-0.9.7/src/diskman/qparted/thread2.cpp
--- diskman-0.9.7-org/src/diskman/qparted/thread2.cpp	2007-11-19 14:49:38.000000000 +0100
+++ diskman-0.9.7/src/diskman/qparted/thread2.cpp	2008-04-27 12:00:16.000000000 +0200
@@ -22,6 +22,7 @@
 #include <iostream>
 #include <sstream>
 #include <string>
+#include <algorithm>
 
 #include <qfile.h>
 #include <qapplication.h>
diff -Naur diskman-0.9.7-org/src/diskman/qparted/thread3.cpp diskman-0.9.7/src/diskman/qparted/thread3.cpp
--- diskman-0.9.7-org/src/diskman/qparted/thread3.cpp	2007-11-19 14:49:38.000000000 +0100
+++ diskman-0.9.7/src/diskman/qparted/thread3.cpp	2008-04-27 12:05:41.000000000 +0200
@@ -22,6 +22,7 @@
 #include <iostream>
 #include <sstream>
 #include <string>
+#include <cstdlib>
 
 #include <qfile.h>
 #include <qapplication.h>
diff -Naur diskman-0.9.7-org/src/diskman/qparted/utils.cpp diskman-0.9.7/src/diskman/qparted/utils.cpp
--- diskman-0.9.7-org/src/diskman/qparted/utils.cpp	2007-11-19 14:49:38.000000000 +0100
+++ diskman-0.9.7/src/diskman/qparted/utils.cpp	2008-04-27 12:10:56.000000000 +0200
@@ -22,6 +22,7 @@
 #include <qvaluevector.h>
 #include <qpainter.h>
 #include <qbitmap.h>
+#include <cstdlib>
 
 #include "qparted/partition.h"
 #include "qparted/utils.h"
diff -Naur diskman-0.9.7-org/src/diskman/qparted/utils.h diskman-0.9.7/src/diskman/qparted/utils.h
--- diskman-0.9.7-org/src/diskman/qparted/utils.h	2007-11-19 14:49:38.000000000 +0100
+++ diskman-0.9.7/src/diskman/qparted/utils.h	2008-04-27 11:45:31.000000000 +0200
@@ -32,6 +32,7 @@
 #include <qpixmap.h>
 
 #include <iostream>
+#include <ctime>
 
 #include "lglobal.h"
 
diff -Naur diskman-0.9.7-org/src/hwprobe/hwprobe.cpp diskman-0.9.7/src/hwprobe/hwprobe.cpp
--- diskman-0.9.7-org/src/hwprobe/hwprobe.cpp	2007-11-19 14:49:41.000000000 +0100
+++ diskman-0.9.7/src/hwprobe/hwprobe.cpp	2008-04-27 11:37:06.000000000 +0200
@@ -235,7 +235,7 @@
     //qDebug( "HwProbePrivate::HwProbePrivate()" );
 
     // create lock file
-    int f = open( "/var/lib/hardware/LOCK", O_CREAT );
+    int f = open( "/var/lib/hardware/LOCK", O_CREAT, S_IRUSR | S_IWUSR );
 
     if( f > 0 )
     {
diff -Naur diskman-0.9.7-org/src/hwprobe/testhwprobe.cpp diskman-0.9.7/src/hwprobe/testhwprobe.cpp
--- diskman-0.9.7-org/src/hwprobe/testhwprobe.cpp	2007-11-19 14:49:41.000000000 +0100
+++ diskman-0.9.7/src/hwprobe/testhwprobe.cpp	2008-04-27 11:37:40.000000000 +0200
@@ -20,6 +20,7 @@
 
 #include <iostream>
 #include <iomanip>
+#include <cstdlib>
 
 #include <foreach.h>
 
