*** AstroMenaceSource/Defines.h	2008-08-14 13:48:12.000000000 +0200
--- AstroMenaceSource/Defines.h.ru	2008-08-14 13:49:09.000000000 +0200
***************
*** 38,46 ****
  // язык интерфейса
  #if !defined(EN) && !defined(DE) && !defined(RU)
  
! #define EN
  //#define DE
! //#define RU
  
  #endif
  
--- 38,46 ----
  // язык интерфейса
  #if !defined(EN) && !defined(DE) && !defined(RU)
  
! //#define EN
  //#define DE
! #define RU
  
  #endif
  
