--- nikto_core.plugin.orig	2009-10-18 00:26:52.000000000 +0200
+++ nikto_core.plugin	2009-11-14 14:05:27.720319756 +0100
@@ -1482,7 +1482,7 @@
   nprint "+ Retrieving 'CHANGES.txt'\n";
   (my $RES, $CONTENT) = fetch("$serverdir/CHANGES.txt","GET");
   if (($CONTENT ne "") && ($RES eq 200)) {
-   open(OUT,">$NIKTOCONFIG{DOCUMENTDIR}/CHANGES.txt") || die print STDERR "+ ERROR: Unable to open '$NIKTOCONFIG{DOCUMENTDIR}/CHANGES.txt' for write: $@\n";
+   open(OUT,">$NIKTOCONFIG{PLUGINDIR}/CHANGES.txt") || die print STDERR "+ ERROR: Unable to open '$NIKTOCONFIG{PLUGINDIR}/CHANGES.txt' for write: $@\n";
    print OUT $CONTENT;
    close(OUT); 
   }
